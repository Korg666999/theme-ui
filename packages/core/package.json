{
  "name": "@theme-ui/core",
  "version": "0.15.0-rc.1",
  "source": "src/index.ts",
  "main": "dist/theme-ui-core.cjs.js",
  "module": "dist/theme-ui-core.esm.js",
  "types": "dist/theme-ui-core.cjs.d.ts",
  "sideEffects": false,
  "scripts": {},
  "repository": "system-ui/theme-ui",
  "author": "Brent Jackson",
  "license": "MIT",
  "publishConfig": {
    "access": "public"
  },
  "dependencies": {
    "@theme-ui/css": "0.15.0-rc.1",
    "deepmerge": "^4.2.2"
  },
  "peerDependencies": {
    "@emotion/react": "^11",
    "react": ">=18"
  },
  "preconstruct": {
    "entrypoints": [
      "index.ts",
      "jsx-dev-runtime.ts",
      "jsx-runtime.ts"
    ]
  }
}
