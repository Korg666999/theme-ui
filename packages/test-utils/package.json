{
  "name": "@theme-ui/test-utils",
  "version": "0.14.5",
  "private": true,
  "license": "MIT",
  "repository": "system-ui/theme-ui",
  "source": "src/index.ts",
  "main": "dist/theme-ui-test-utils.cjs.js",
  "module": "dist/theme-ui-test-utils.esm.js",
  "types": "dist/theme-ui-test-utils.cjs.d.ts",
  "sideEffects": false,
  "scripts": {
    "build": "tsc",
    "watch": "tsc -w",
    "typecheck": "tsc --noEmit"
  },
  "dependencies": {
    "@testing-library/react": "^13.3.0",
    "conditional-type-checks": "^1.0.6",
    "react-test-renderer": "^18.2.0",
    "ts-snippet": "^5.0.0"
  },
  "peerDependencies": {
    "react": ">=18",
    "theme-ui": ">=0.14.0"
  },
  "devDependencies": {
    "react": "^18",
    "theme-ui": "^0.14.5"
  }
}
