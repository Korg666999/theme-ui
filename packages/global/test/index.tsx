/**
 * @jest-environment jsdom
 * @jsx jsx
 */

import { jsx } from '@theme-ui/core'
import { cleanup } from '@testing-library/react'
import { render } from '@theme-ui/test-utils'
import { matchers } from '@emotion/jest'

import { ThemeProvider } from '@theme-ui/core'
import Global from '../src'

expect.extend(matchers)

afterEach(cleanup)

test('renders global styles', async () => {
  const root = render(
    <ThemeProvider
      theme={{
        config: {
          useRootStyles: false,
        },
        fonts: {
          body: 'Georgia,serif',
        },
      }}
    >
      <Global
        sx={{
          '@font-face': {
            fontFamily: 'some-name',
          },
          h1: {
            color: 'salmon',
            fontFamily: 'body',
          },
          body: {
            margin: 0,
          },
        }}
      />
      <h1>Hello</h1>
    </ThemeProvider>
  )

  const style = window.getComputedStyle(root.baseElement)
  expect(style.margin).toBe('0px')

  // const h1 = root.baseElement.querySelector('h1')!
  // const style = global.getComputedStyle(h1)
  // console.log(style)
  // expect(style.fontFamily).toBe('Georgia,serif')
  // expect(style.color).toBe('salmon')
})
