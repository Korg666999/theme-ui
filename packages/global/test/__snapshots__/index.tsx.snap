// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders global styles 1`] = `
<h1>
  Hello
</h1>
`;
